lightly/api/upload.py

```diff
@@ -187,22 +79,22 @@ def _upload_single_image(image,
     # upload sample with metadata
     sample_upload_success = True
 
+    api_client = get_api_client(token=token)
+    samples_api = SamplesApi(api_client=api_client)
+
     try:
-        sample_id = routes.users.datasets.samples.post(
-            basename, thumbname, metadata, dataset_id, token
-        )
-    except RuntimeError as e:
-        sample_upload_success = False
-        raise ValueError
+        body = SampleCreateRequest(file_name=basename, thumb_name=thumbname, meta_data=metadata)
+        sample_id = samples_api.create_sample_by_dataset_id(body=body, dataset_id=dataset_id).id
+    except RuntimeError:
+        raise ValueError("Creating the sampling in the web platform failed.")
 
     # upload thumbnail
     thumbnail_upload_success = True
     if mode == 'thumbnails' and not metadata['is_corrupted'] and sample_upload_success:
         try:
             # try to get signed url for thumbnail
-            signed_url = routes.users.datasets.samples. \
-                get_presigned_upload_url(
-                    thumbname, dataset_id, sample_id, token)
+            signed_url = samples_api.\
+                get_sample_image_write_url_by_id(dataset_id=dataset_id, sample_id=sample_id, is_thumbnail=True)
             # try to upload thumbnail
             upload_file_with_signed_url(
                 PIL_to_bytes(thumbnail, ext='webp', quality=70),
```

Comment:
Oh, here we need to improve quality to 90 to match the thumbnail generator in the API (my bad). Can you fix this please?